features/cellular/easy_cellular/EasyCellularConnection.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2017, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CellularTargets.h"
+#ifdef CELLULAR_DEVICE
+ 
+#if NSAPI_PPP_AVAILABLE
+#include "nsapi_ppp.h"
+#endif
+
+#include "CellularConnectionUtil.h"
+#include "CellularUtil.h"
+
+#include "EasyCellularConnection.h"
+
+#include "CellularLog.h"
+
+namespace mbed {
+
+static CellularConnectionUtil cellularConnection;
+static rtos::Semaphore cellularSemaphore(0);
+static UARTSerial cellularSerial(MDMTXD, MDMRXD, MBED_CONF_PLATFORM_DEFAULT_SERIAL_BAUD_RATE);
```

Comment:
You can't just define these objects as static - they will be constructed and included in the image even if an `EasyCellularConnection` is never used. 

Make them normal class members to avoid this issue.

Reply:
👍 or use SingletonPtr class

Reply:
Changed as class member variables